frontend/packages/console-shared/src/hooks/plugins-overview-tab-section.ts

```diff
@@ -11,10 +11,15 @@ export const usePluginsOverviewTabSection = (
   item: OverviewItem,
 ): { Component: React.FC<AsyncComponentProps>; key: string }[] => {
   const tabSections = useExtensions(isOverviewTabSection);
-  return tabSections
-    .filter((section) => item[section.properties.key])
-    .map((section: OverviewTabSection) => ({
-      Component: getResourceTabSectionComp(section),
-      key: section.properties.key,
-    }));
+  return React.useMemo(
+    () =>
+      tabSections
+        .filter((section) => item[section.properties.key])
+        .map((section: OverviewTabSection) => ({
+          Component: getResourceTabSectionComp(section),
+          key: section.properties.key,
+        })),
+    // `item` is complex object but we only use the presence of keys as a dependency
+    [JSON.stringify(Object.keys(item).sort())],
```

Comment:
@christianvogt @sahil143 Sorting the dep keys and stringinfying it to make sure the size doesn't change even if there is additional keys added dynmaically and  string comparision should work instead of having custom comparator. WDYT?

Reply:
this works fine, I am not seeing warning anymore, tested locally.